### PR TITLE
#127 - Send all data via rabbit to API

### LINKED ISSUE #127: Create mechanism to pass data to the API
Once the API is more mature, we will need to be able to pass such data to the API.

### PR DESCRIPTION
**Pull request by Maelstromeous**
Closes #127

### CODE REVIEW

File: src/data/ApiMQMessage.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types,@typescript-eslint/no-explicit-any */
+export enum ApiMQOperations {
+    CREATE = 'create',
+    UPDATE = 'update',
+    DELETE = 'delete',
+}
```
Comment (Maelstromeous) [inline src/data/ApiMQMessage.ts:6]: Remove, not used

File: src/services/rabbitmq/RabbitMQConnectionHandlerFactory.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import {ConfirmChannel} from 'amqplib';
+import {getLogger} from '../../logger';
+import RabbitMQ from '../../config/rabbitmq';
+import ApplicationException from '../../exceptions/ApplicationException';
+import {injectable} from 'inversify';
+
+@injectable()
+export class RabbitMQConnectionHandlerFactory {
+    private static readonly logger = getLogger('RabbitMQConnectionHandler');
+
```
Comment (Maelstromeous) [inline src/services/rabbitmq/RabbitMQConnectionHandlerFactory.ts:11]: Space

File: src/services/rabbitmq/subscribers/AdminAggregatorSubscriber.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import ApplicationException from '../../../exceptions/ApplicationException';
+
+@injectable()
+export default class AdminAggregatorSubscriber implements RabbitMQConnectionAwareInterface {
+    private static readonly logger = getLogger('AdminAggregatorSubscriber');
+    private static channelWrapper: ChannelWrapper;
+    private static queueName = '';
+    private static adminMessageHandlers: Array<MessageQueueHandlerInterface<ParsedQueueMessage>>;
+    private readonly connectionHandlerFactory: RabbitMQConnectionHandlerFactory;
+    constructor(
```
Comment (Maelstromeous) [inline src/services/rabbitmq/subscribers/AdminAggregatorSubscriber.ts:22]: Need line

File: src/services/subscribers/CensusEventSubscriberService.ts [hunk 4]
```diff
@@ -99,6 +68,18 @@ export default class CensusEventSubscriberService implements ServiceInterface {
         CensusEventSubscriberService.logger.debug('Terminating Census Stream Service!');
     }
 
+    private static handleCharacterException(service: string, message: string): void {
```
Comment (Maelstromeous) [inline src/services/subscribers/CensusEventSubscriberService.ts:71]: This was moved

File: src/services/subscribers/CensusEventSubscriberService.ts [hunk 5]
```diff
@@ -189,53 +170,6 @@ export default class CensusEventSubscriberService implements ServiceInterface {
...
-            if (!character) {
-                return;
-            }
-
-            const playerLogoutEvent = new PlayerLogoutEvent(event, character);
-            await this.playerLogoutEventHandler.handle(playerLogoutEvent);
-        });
-    }
-
-    private handleCharacterException(service: string, message: string): void {
```
Comment (Maelstromeous) [inline src/services/subscribers/CensusEventSubscriberService.ts:230]: This was moved, rest deleted

**Comment by microwavekonijn** [comment]
Working on https://github.com/microwavekonijn/ps2census-nestjs. Should simplify things a lot and allows us to use the functionality of pushing events and request to the message queue from nestjs.

**Comment by Maelstromeous** [comment]
Awaiting completion of #169

**Pull request merged**